libs/wallet/src/connectors/json-rpc-connector.ts

```diff
@@ -0,0 +1,296 @@
+import { z } from 'zod';
```

Comment:
not sure if interested in anything like this, in the past days i had to do something similar for the wallet desktop, and ended up pulling in a jsonrpc generator. 
this was my template: https://github.com/vegaprotocol/vegawallet-desktop/pull/307/files#diff-f285e942b949fd0bea71f13e8e3814ea2d892a809696ae690a4d4e843cba71d2
it's pretty neat, because it can go and grab a remote openrpc spec and generate all the types for the API you're interacting with, and the `MethodCallValidator` can do runtime checks if needed. you can generate a custom client / connector / anything you might need by writing a template. imho it is worth checking out.

Reply:
Cool, thanks for the heads up. I recently got bitten by using a generator for the v1 service, it was a bit premature and it ended up being too much of a manual maintenance job as there was no way to generate a spec. It ended up just being a massive pain, hence pulling it out and doing something more manual. That said I think v2 is a lot more solid and it probably makes more sense to use something like this now. Lets consider it for some future work